baseServer_test.go

```diff
@@ -242,37 +274,42 @@ func TestWaitForMessagingToEnd(t *testing.T) {
 	const numAgents = 10
 	m := make([]basePlatformSOMAS.AgentGeneratorCountPair[ITestBaseAgent], 1)
 	m[0] = basePlatformSOMAS.MakeAgentGeneratorCountPair(NewTestAgent, numAgents)
-	server := basePlatformSOMAS.CreateServer(m, 1, 1, time.Second)
+	server := basePlatformSOMAS.CreateServer(m, 1, 1, time.Second, time.Millisecond)
 	agentMap := server.GetAgentMap()
 	arrayOfIDs := make([]uuid.UUID, numAgents)
 	i := 0
 	for id, ag := range agentMap {
 		arrayOfIDs[i] = id
 		i++
-		ag.SetGoal(numberOfMessages *(numAgents))
+		ag.SetGoal(numberOfMessages * (numAgents))
 	}
 
 	for j := 0; j < numberOfMessages; j++ {
-		for _,ag := range server.GetAgentMap() {
-		msg := ag.NewTestMessage()
-		//server.IncrementWaitGroup()
-		go ag.SendMessage(msg, arrayOfIDs)
-	}}
-	server.EndAgentListeningSession()
+		for _, ag := range server.GetAgentMap() {
+			msg := ag.NewTestMessage()
+			go ag.SendMessage(msg, arrayOfIDs)
+		}
+	}
+	a, b := server.EndAgentListeningSession()
+	if !b {
+		t.Error(a)
+	}
+	//fmt.Println(a,b)
 	for _, ag := range agentMap {
-
 		if !ag.ReceivedMessage() {
 			t.Errorf("agent %s recieved %d messages, expected %d\n", ag.GetID(), ag.GetCounter(), ag.GetGoal())
+		} else {
+			fmt.Printf("agent %s recieved %d messages, expected %d\n", ag.GetID(), ag.GetCounter(), ag.GetGoal())
+
```

Comment:
why are we printing to console in a test? You already have the counter case handled by the error

Reply:
when one agents doesnt get all of its messages it is useful to see the message status of the rest of the agents